integration/valkey_search_test_case.py

```diff
@@ -54,3 +58,156 @@ def parse_valkey_info(self, section):
                 key, value = line.split(":", 1)
                 stats_dict[key.strip()] = value.strip()
         return stats_dict
+
+
+class ValkeySearchClusterTestCase(ValkeySearchTestCaseBase):
+    def _start_server(self, port, test_name):
+        server_path = os.getenv("VALKEY_SERVER_PATH")
+        testdir_base = f"/tmp/valkey-search-clusters/{test_name}"
+        testdir = f"{testdir_base}/{port}"
+
+        os.makedirs(testdir, exist_ok=True)
+        curdir = os.getcwd()
+        os.chdir(testdir)
+        lines = [
+            "enable-debug-command yes",
+            f"loadmodule {os.getenv('JSON_MODULE_PATH')}",
+            f"loadmodule {os.getenv('MODULE_PATH')} --use-coordinator",
+            f"dir {testdir}",
+            "cluster-enabled yes",
+        ]
+
+        conf_file = f"{testdir}/valkey.conf"
+        with open(conf_file, "w+") as f:
+            for line in lines:
+                f.write(f"{line}\n")
+            f.write("\n")
+            f.close()
+
+        server, client = self.create_server(
+            testdir=testdir,
+            server_path=server_path,
+            args="",
+            port=port,
+            conf_file=conf_file,
+        )
+        os.chdir(curdir)
+        logfile = f"{testdir}/logfile_{port}"
+        return server, client, logfile
+
+    def _split_range_pairs(self, start, end, n):
+        points = [start + i * (end - start) // n for i in range(n + 1)]
+        return list(zip(points[:-1], points[1:]))
+
+    def add_slots(self, node_idx, first_slot, last_slot):
+        client: Valkey = self.client_for_primary(node_idx)
+        slots_to_add = list(range(int(first_slot), int(last_slot)))
+        try:
+            client.execute_command("CLUSTERADMIN ADDSLOTS", *slots_to_add)
+        except Exception as e:
+            client.execute_command("CLUSTER ADDSLOTS", *slots_to_add)
+
+    def cluster_meet(self, node_idx, primaries_count):
+        client: Valkey = self.client_for_primary(node_idx)
+
+        for node_to_meet in range(0, primaries_count):
+            if node_to_meet == node_idx:
+                continue
+            try:
+                client.execute_command(
+                    " ".join(
+                        [
+                            "CLUSTERADMIN",
```

Comment:
There is a typo here likely 

```suggestion
                            "CLUSTER",
```